casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -324,20 +324,22 @@ class RuntimeManagerImpl[F[_]: Concurrent: Metrics: Span: Log](
                     and event logs, so they cannot be reliably compared. */
                  else if (errors.nonEmpty)
                    runtime.replaySpace
-                     .checkReplayData()
-                     .attempt
-                     .flatMap {
-                       case Right(_) => none[ReplayFailure].pure[F]
-                       case Left(ex: ReplayException) => {
-                         Log[F].error(s"Failed during processing of deploy: ${processedDeploy}") >>
-                           (none[DeployData], UnusedCommEvent(ex): Failed).some
-                             .pure[F]
-                       }
-                       case Left(ex) =>
-                         (none[DeployData], UserErrors(Vector(ex)): Failed).some
-                           .pure[F]
-                     }
-                 }
+                     .revertToSoftCheckpoint(softCheckpoint)
```

Comment:
This looks like an unrelated change for `Simplify syntax in `replayComputeState`.`, so I'm not sure if it's correct.

It also seems that we should revert to soft-checkpoint on cost discrepancy and event log discrepancy as well - currently we do so only for rholang execution errors.

Reply:
This was caused by a bad merge. Good catch!